### PR TITLE
4 - View a list of all articles

### PR DESCRIPTION
**Pull request by Tiny-B**
This pull request will add a home page with a flex box container holding a list of all articles sorted by date, the container and items scale with screen size using a media query in css.

It will also add basic style to the home page with a side bar that opens and closes to reveal extra buttons, the logo will also act as a link to home and is present at all times.

### CODE REVIEW

File: src/api.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import axios from 'axios';
+
+const newsAPI = axios.create({
+	baseURL: 'https://nc-news-xd69.onrender.com/api',
+});
+
+const fetchArticles = async () => {
+	const results = await newsAPI.get('/articles');
+	const sortedResults = results.data.articles.sort(
```
Comment (callumu21) [inline src/api.js]: We shouldn't be sorting with array methods when our endpoint can accept a sort query for data. It's more efficient, and this wouldn't be sustainable if we were retrieving large amounts of data. Please change this to use a query instead

File: src/components/home-components/ArticleCard.jsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+		<article className='article-card'>
+			<img
+				src={article.article_img_url}
+				alt={article.topic}
+			/>
+			<div className='text-content'>
+				<h4>{article.title}</h4>
+				<h5>{article.author}</h5>
+				<p>
+					{new Date(article.created_at)
```
Comment (callumu21) [inline src/components/home-components/ArticleCard.jsx:12]: Nice job considering how users will want to read date here

File: src/components/home-components/ArticleContainer.jsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import ArticleCard from './ArticleCard';
+import { api } from '../../api';
+import { useEffect, useState } from 'react';
+
+const ArticleContainer = () => {
+	const [articles, setArticles] = useState([]);
+
+	useEffect(() => {
+		api.fetchArticles().then(res => {
+			// console.log(res.data.articles[0]);
```
Comment (callumu21) [inline src/components/home-components/ArticleContainer.jsx]: We can remove this console.log now

File: src/components/home-components/ArticleContainer.jsx [hunk 4]
```diff
@@ -0,0 +1,29 @@
+import ArticleCard from './ArticleCard';
+import { api } from '../../api';
+import { useEffect, useState } from 'react';
+
+const ArticleContainer = () => {
+	const [articles, setArticles] = useState([]);
+
+	useEffect(() => {
```
Comment (callumu21) [inline src/components/home-components/ArticleContainer.jsx:9]: We should have some kind of loading state here and be conditionally rendering a loading message to users with a slower connection

**Review by callumu21 — CHANGES_REQUESTED** [review]
Great start here! Nice to see some solid semantic HTML choices, and the start of responsiveness across your web app. I've left a few comments for you to consider, but the key thing missing at the moment is:

1. Loading state for the ArticleContainer component. Users should be able to see a loading message while the fetch request is still pending.

Once you've added that in, please resubmit this PR and we can take another look 👍

**Review by Pipe-Smoking-Rabbit — APPROVED** [review]
Smashed it Kief! 👍

**Pull request merged**